api/src/test/java/org/pmiops/workbench/actionaudit/adapters/AuthDomainAuditAdapterTest.java

```diff
@@ -0,0 +1,89 @@
+package org.pmiops.workbench.actionaudit.adapters;
+
+import static com.google.common.truth.Truth.assertThat;
+import static org.mockito.Mockito.verify;
+
+import java.time.Clock;
+import java.time.Instant;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Captor;
+import org.pmiops.workbench.actionaudit.ActionAuditEvent;
+import org.pmiops.workbench.actionaudit.ActionAuditService;
+import org.pmiops.workbench.actionaudit.ActionType;
+import org.pmiops.workbench.actionaudit.AgentType;
+import org.pmiops.workbench.actionaudit.TargetType;
+import org.pmiops.workbench.actionaudit.targetproperties.AccountTargetProperty;
+import org.pmiops.workbench.db.model.DbUser;
+import org.pmiops.workbench.test.FakeClock;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.config.ConfigurableBeanFactory;
+import org.springframework.boot.autoconfigure.liquibase.LiquibaseAutoConfiguration;
+import org.springframework.boot.test.context.TestConfiguration;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Import;
+import org.springframework.context.annotation.Scope;
+import org.springframework.test.context.junit4.SpringRunner;
+
+@RunWith(SpringRunner.class)
+@Import(LiquibaseAutoConfiguration.class)
+public class AuthDomainAuditAdapterTest {
+
+  private static final long USER_ID = 101L;
+  private static final long ADMINISTRATOR_USER_ID = 222L;
+  private static final String ADMINISTRATOR_EMAIL = "admin@aou.biz";
+  private static final Instant INSTANT = Instant.parse("2000-01-01T00:00:00.00Z");
+  private static final String ACTION_ID = "9095d2f9-8db2-46c3-8f8e-4f90a62b457f";
+
+  @Autowired private ActionAuditService mockActionAuditService;
+
+  @Captor private ArgumentCaptor<ActionAuditEvent> eventCaptor;
+
+  @Autowired private AuthDomainAuditAdapter authDomainAuditAdapter;
+
+  @TestConfiguration
+  @Import({AuthDomainAuditAdapterImpl.class})
+  @MockBean({ActionAuditService.class})
+  static class Config {
+    @Bean(name = "ACTION_ID")
+    @Scope(value = ConfigurableBeanFactory.SCOPE_PROTOTYPE)
```

Comment:
Scope technically isn't necessary since a single constant value is being returned. You could also just inline the value here instead of referencing a static variable.

Reply:
I wanted to match the scope on the real Bean, which delivers a different one each time. From the perspective of this test as it exists now, you're right, but I wanted to reserve the right to change the value in between calls in the same test.